File: 189-rotate-array/rotate-array.cpp
class Solution {
public:
void print(vector<int> &nums)
{
    for(int i=0;i<nums.size();i++)
    {
        cout<<nums[i];
    }
    cout<<endl;
}
    void rotate(vector<int>& nums, int k) {
       int n=nums.size();
       k =k%n;
    //    if(n<k)
    //    {
    //     return ;
    //    }
       reverse(nums.begin(),nums.end()-k);
       print(nums);
       reverse(nums.begin()+n-k,nums.end());
       print(nums);
       reverse(nums.begin(),nums.end());
    }
};

File: 189-rotate-array/rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k = k%n
        if(k>n or k<0):
            k = 1
        nums[:]=nums[n-k:]+nums[:n-k]
        